### PR TITLE
Add glslls

### PR DESCRIPTION
**Pull request by svenstaro**
This adds a [glslls](https://github.com/svenstaro/glsl-language-server)-based linter.

### CODE REVIEW

File: ale_linters/glsl/glslls.vim [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+call ale#Set('glsl_glslls_executable', 'glslls')
+
+function! ale_linters#glsl#glslls#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'glsl_glslls_executable')
+endfunction
+
+function! ale_linters#glsl#glslls#GetCommand(buffer) abort
+    let l:executable = ale_linters#glsl#glslls#GetExecutable(a:buffer)
+
+    return ale#Escape(l:executable) . ' -l ' . tempname() . ' --stdin --verbose'
```
Comment (w0rp) [inline ale_linters/glsl/glslls.vim]: Could you make logging with `-l` and `--verbose` optional?

File: ale_linters/glsl/glslls.vim [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+endfunction
+
+function! ale_linters#glsl#glslls#GetLanguage(buffer) abort
+    return 'glsl'
+endfunction
+
+function! ale_linters#glsl#glslls#GetProjectRoot(buffer) abort
+    let l:git_path = ale#path#FindNearestDirectory(a:buffer, '.git')
```
Comment (w0rp) [inline ale_linters/glsl/glslls.vim]: Use the `ale#c#FindProjectRoot` function here instead, which looks for a few different files.

**Comment by svenstaro** [comment]
Alright, I've addressed your stuff @w0rp.

File: test/command_callback/test_glslls_command_callback.vader [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Execute(Executable should be configurable):
+  let g:ale_glsl_glslls_executable = 'foobar'
+  AssertEqual
+  \ 'foobar',
+  \ ale_linters#glsl#glslls#GetExecutable(bufnr(''))
+
+Execute(Command should use executable):
+  let command1 = ale_linters#glsl#glslls#GetCommand(bufnr(''))
+  AssertEqual command1, "'glslls' --stdin"
```
Comment (w0rp) [inline test/command_callback/test_glslls_command_callback.vader]: Use `ale#Escape` on these test values instead. Escaping works differently on Windows, and the tests are failing in AppVeyor. Prefer double quoted strings. You can escape single quotes in single quoted strings by doubling them, but you won't need to do that here.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**